### PR TITLE
[14.0][MIG] sale_automatic_workflow: Migration to 14.0

### PR DESCRIPTION
**Pull request by britoederr**
Standard migration...

### CODE REVIEW

**Review by danimv5 — APPROVED** [review]
Functional review carried out both locally and in the runbot. It works well. LGTM

**Review by dnplkndll — APPROVED** [review]
looks good, Identical to the PR submitted in duplicate. thanks

**Comment by dnplkndll** [comment]
@max3903 @SodexisTeam please review.

**Comment by bodedra** [comment]
@britoederr run pre-commit to make CI happy.

File: sale_automatic_workflow/models/automatic_workflow_job.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    @api.model
+    def _validate_invoices(self, validate_invoice_filter):
+        move_obj = self.env["account.move"]
+        invoices = move_obj.search(validate_invoice_filter)
+        _logger.debug("Invoices to validate: %s", invoices.ids)
+        for invoice in invoices:
+            with savepoint(self.env.cr), force_company(self.env, invoice.company_id):
+                # FIX Why is this needed or certain invoices
+                # in enterprise in multicompany?
+                invoice.with_context(force_company=invoice.company_id.id).post()
```
Comment (atchuthan) [inline sale_automatic_workflow/models/automatic_workflow_job.py:76]: Odoo shows Deprecation Warning in log
RedirectWarning method 'post()' is a deprecated alias to 'action_post()' or _post()

Also, Odoo shows logger warning 
Context key 'force_company' is no longer supported. Use with_company(company) instead.

**Review by atchuthan — CHANGES_REQUESTED** [review]

File: sale_automatic_workflow/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    float_compare(
+                        move.quantity_done,
+                        move.product_qty,
+                        precision_rounding=rounding,
+                    )
+                    == -1
+                ):
+                    for move_line in move.move_line_ids:
+                        move_line.qty_done = move_line.product_uom_qty
+            picking.with_context(skip_overprocessed_check=True).button_validate()
```
Comment (florian-dacosta) [inline sale_automatic_workflow/models/stock_picking.py:36]: Context `skip_overprocessed_check` does not exist anymore in stock module. I think `skip_immediate` should be used instead

**Comment by atchuthan** [comment]
Superseded by https://github.com/OCA/sale-workflow/pull/1397
Please review this PR and let's try to merge this one with all the changes requested in this PR

**Comment by rousseldenis** [comment]
Closing this in favor of #1397

**Pull request closed without merging**